lib/uri/common.rb

```diff
@@ -54,8 +54,7 @@ def make_components_hash(klass, array_hash)
         raise ArgumentError,
           "expected Array of or Hash of components of #{klass} (#{klass.component[1..-1].join(', ')})"
       end
-      tmp[:scheme] = klass.to_s.sub(/\A.*::/, '').downcase
-
+      tmp[:scheme] = klass.to_s.sub(/\A.*::/, '').tap(&:downcase!)
```

Comment:
You could use ``(tmp[:scheme] = klass.to_s.sub(/\A.*::/, '')).downcase!``, but this one is a weird little trick to avoid tap.
I don't know if you can apply ``sub!`` here, but if you can the following could be better:
```ruby
scheme = tmp[:scheme] = klass.to_s
scheme.sub!(/\A.*::/, '')
scheme.downcase!
```

Reply:
It can work as long as `to_s` returns a new string with every call.